crates/driver/src/domain/competition/solution/settlement.rs

```diff
@@ -415,6 +396,27 @@ impl Gas {
     /// The balance required to ensure settlement execution with the given gas
     /// parameters.
     pub fn required_balance(&self) -> eth::Ether {
-        self.limit * self.price
+        self.limit * self.fee_per_gas()
+    }
+
+    // Compute an upper bound for `max_fee_per_gas` for the given settlement.
```

Comment:
nit: To make it show up as a comment when using an LSP or an IDE.
```suggestion
    /// Compute an upper bound for `max_fee_per_gas` for the given settlement.
```